framework/src/base/Assembly.C

```diff
@@ -570,51 +578,53 @@ Assembly::reinitNeighbor(const Elem * neighbor, const std::vector<Point> & refer
   _current_neighbor_elem = neighbor;
 
   // Calculate the volume of the neighbor
-
-  FEType fe_type (neighbor->default_order() , LAGRANGE);
-  std::unique_ptr<FEBase> fe (FEBase::build(neighbor->dim(), fe_type));
-
-  const std::vector<Real> & JxW = fe->get_JxW();
-  const std::vector<Point> & q_points = fe->get_xyz();
-
-  // The default quadrature rule should integrate the mass matrix,
-  // thus it should be plenty to compute the area
-  QGauss qrule (neighbor->dim(), fe_type.default_quadrature_order());
-  fe->attach_quadrature_rule(&qrule);
-  fe->reinit(neighbor);
-
-  // set the coord transformation
-  MooseArray<Real> coord;
-  coord.resize(qrule.n_points());
-  Moose::CoordinateSystemType coord_type = _sys.subproblem().getCoordSystem(neighbor->subdomain_id());
-  unsigned int rz_radial_coord = _sys.subproblem().getAxisymmetricRadialCoord();
-  switch (coord_type) // coord type should be the same for the neighbor
+  if (_need_neighbor_elem_volume)
   {
-    case Moose::COORD_XYZ:
-      for (unsigned int qp = 0; qp < qrule.n_points(); qp++)
-        coord[qp] = 1.;
-      break;
-
-    case Moose::COORD_RZ:
-      for (unsigned int qp = 0; qp < qrule.n_points(); qp++)
-        coord[qp] = 2 * M_PI * q_points[qp](rz_radial_coord);
-      break;
+    FEType fe_type(neighbor->default_order(), LAGRANGE);
```

Comment:
This is why this is showing up in timing... there are a BUNCH of temporaries here.  These should all get moved to member variables.